src/main/java/core/basesyntax/store/io/reader/FileReaderImpl.java

```
@@ -0,0 +1,19 @@
+package core.basesyntax.store.io.reader;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+
+public class FileReaderImpl implements FileReader {
+    @Override
+    public List<String> read(String fileName) {
+        try {
+            return Files.readAllLines(Path.of(fileName));
+        } catch (IOException e) {
+            e.printStackTrace();
+            return Collections.emptyList();
```

Consider using a logging framework instead of `printStackTrace()` for better control over error logging. Additionally, returning an empty list might not be the best way to handle errors, as it could lead to silent failures. Consider throwing a custom exception or handling the error in a way that informs the caller of the failure.